src/H5Dchunk.c

```diff
@@ -1205,6 +1205,10 @@ H5D__chunk_io_init_selections(H5D_io_info_t *io_info, H5D_dset_io_info_t *dinfo)
         /* Initialize skip list for chunk selections */
         fm->use_single = true;
 
+        /* Initialize selection type in memory and file */
```

Comment:
I suggest that you put this in H5D__chunk_io_init() instead, so that it doesn't get missed for another case in the future.

Reply:
I see we are both working this evening :)